chat-sdk-ui/src/main/java/co/chatsdk/ui/ImageListAdapter.java

```diff
@@ -0,0 +1,107 @@
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+
+public class ImageListAdapter extends RecyclerView.Adapter<ImageListAdapter.MyViewHolder>  {
+
+    public ArrayList<Uri> uriArrayList = new ArrayList<>();
+    private WeakReference<SendMultipleImagesAtOnceActivity> theMainActivity;
+
+    public ImageListAdapter (SendMultipleImagesAtOnceActivity activity) {
+        this.theMainActivity = new WeakReference<SendMultipleImagesAtOnceActivity>(activity);
+    }
+
+    public class MyViewHolder extends RecyclerView.ViewHolder {
+
+        private WeakReference<ImageListAdapter> adapter;
+        public ImageView imageView;
+        public Uri thisImageUri;
+        public int thisImagePosition;
+
+        public MyViewHolder(ImageListAdapter la, ImageView v) {
+            super(v);
+            this.adapter = new WeakReference<ImageListAdapter>(la);
+            imageView = v;
+
+            imageView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    setMainImageDisplay(thisImageUri);
+                    setMainImageUri(thisImageUri);
+                }
+            });
+
+            imageView.setOnLongClickListener(new View.OnLongClickListener() {
+                @Override
+                public boolean onLongClick(View v) {
+                    Uri toBeDeletedUri = thisImageUri;
+                    removeImageFromAdapter(thisImageUri);
+                    notifyDataSetChanged();
+                    if (getMainImageUri() == toBeDeletedUri) {
+                        if (uriArrayList.size() >= 1 && uriArrayList.size() == thisImagePosition) {
+                            setMainImageDisplay(uriArrayList.get(thisImagePosition - 1));
+                            setMainImageUri(uriArrayList.get(thisImagePosition - 1));
+                        }
+                        else if (uriArrayList.size() > 1) {
+                            setMainImageDisplay(uriArrayList.get(thisImagePosition));
+                            setMainImageUri(uriArrayList.get(thisImagePosition));
+                        }
+                    }
+                    return true;
+                }
+            });
+        }
+    }
+
+    @Override
+    public ImageListAdapter.MyViewHolder onCreateViewHolder (ViewGroup parent, int viewType) {
+
+        ImageView v = (ImageView) LayoutInflater.from(parent.getContext()).inflate(R.layout.image_view, parent, false);
+        MyViewHolder vh = new MyViewHolder(this, v);
+        return vh;
+    }
+
+    @Override
+    public void onBindViewHolder(MyViewHolder holder, int position) {
+        holder.imageView.setImageURI(uriArrayList.get(position));
+        holder.thisImageUri = uriArrayList.get(position);
+        holder.thisImagePosition = position;
```

Comment:
This can be simplified. Make a method on the holder called `setURI(` and then pass in the URI. Then the holder would set it's class URI property and add the URI to the imageView.

Reply:
OK, but the method onBindViewHolder() must exists in this class, I put the method I just amde inside of the onBindViewHolder method.